### PR TITLE
allows multiple desktop creations

### PR DESCRIPTION
**Pull request by jvinolas**
new 'bulk desktops' button created at domains page. It allows to create bulk desktops to individual users or users in categories, groups and roles.

### CODE REVIEW

**Review by sim6 — CHANGES_REQUESTED** [review]
I'm not inspected the code but it works for me. Please, adapt https://github.com/isard-vdi/isard/blob/develop/webapp/webapp/webapp/static/js/snippets/alloweds.js#L74 and https://github.com/isard-vdi/isard/blob/develop/webapp/webapp/webapp/static/js/snippets/alloweds.js#L104 to show a better placeholder for fields of "Users to create this desktop" form section.

**Review by sim6 — CHANGES_REQUESTED** [review]
Works for me. But:

1. Placeholder is not clear, see https://github.com/isard-vdi/isard/pull/354#pullrequestreview-555925341
2. As manager, the modal isn't closed after click Create desktop(s) button.
3. If there is any error, like the user have a desktop with the same name, no error is shown.
4. Please, squash commits.

**Review by NefixEstrada — APPROVED** [review]

File: webapp/webapp/webapp/static/admin/js/domains.js [hunk 1]
```diff
@@ -38,7 +38,72 @@ $(document).ready(function() {
...
+                    text: "Can't create another desktop, category quota full.",
+                    hide: true,
+                    delay: 3000,
+                    icon: 'fa fa-alert-sign',
+                    opacity: 1,
+                    type: 'error'
+                });                
+        }else{	
+            setHardwareOptions('#modalAddDesktop');
+            //~ setHardwareDomainDefaults('#modalAddDesktop',pk);
```
Comment (sim6) [inline webapp/webapp/webapp/static/admin/js/domains.js:95]: What about this comment?
```suggestion
```

File: webapp/webapp/webapp/static/admin/js/domains.js [hunk 2]
```diff
@@ -411,7 +476,7 @@ $(document).ready(function() {
             $("#modalAddFromMedia #modalAdd")[0].reset();
             $("#modalAddFromMedia").modal('hide');   
             $("#modalTemplateDesktop #modalTemplateDesktopForm")[0].reset();
-            $("#modalTemplateDesktop").modal('hide');             
+            $("#modalTemplateDesktop").modal('hide');
```
Comment (sim6) [inline webapp/webapp/webapp/static/admin/js/domains.js:479]: Could you remove the tailing whitespaces or not touch this line?
```suggestion
            $("#modalTemplateDesktop").modal('hide');
```

File: webapp/webapp/webapp/static/admin/js/domains.js [hunk 3]
```diff
@@ -421,9 +486,28 @@ $(document).ready(function() {
                 icon: 'fa fa-'+data.icon,
                 opacity: 1,
                 type: data.type
-        });
+        });
```
Comment (sim6) [inline webapp/webapp/webapp/static/admin/js/domains.js:489]: Please, could you remove tailing whitespaces?
```suggestion
        });
```

File: webapp/webapp/webapp/static/js/snippets/desktop_add.js [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+			"ajax": {
+				"url": "/isard-admin/desktops/getAllTemplates/",
+				"dataSrc": ""
+			},
+
+            "scrollY":        "125px",
+            "scrollCollapse": true,
+            "paging":         false,
+            
+            //~ "searching":         false,
```
Comment (sim6) [inline webapp/webapp/webapp/static/js/snippets/desktop_add.js:36]: What about this comment?
```suggestion
```

File: webapp/webapp/webapp/static/js/snippets/desktop_add.js [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+			"rowId": "id",
+			"deferRender": true,
+			"columns": [
+				//~ {
+                //~ "className":      'details-control',
+                //~ "orderable":      false,
+                //~ "data":           null,
+                //~ "width": "10px",
+                //~ "defaultContent": '<button class="btn btn-xs btn-info" type="button"  data-placement="top" ><i class="fa fa-plus"></i></button>'
+				//~ },
```
Comment (sim6) [inline webapp/webapp/webapp/static/js/snippets/desktop_add.js:53]: Why not remove this comments?
```suggestion
```

File: webapp/webapp/webapp/static/js/snippets/desktop_add.js [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+                        //~ .appendTo( $(column.header()).empty() )
+                        //~ .on( 'change', function () {
+                            //~ var val = $.fn.dataTable.util.escapeRegex(
+                                //~ $(this).val()
+                            //~ );
+                            //~ column
+                                //~ .search( val ? '^'+val+'$' : '', true, false )
+                                //~ .draw();
+                        //~ } );
+        //~ }
```
Comment (sim6) [inline webapp/webapp/webapp/static/js/snippets/desktop_add.js:23]: Why this commented block?
```suggestion
```

File: webapp/webapp/webapp/static/admin/js/domains.js [hunk 7]
```diff
@@ -38,7 +38,72 @@ $(document).ready(function() {
...
+            new PNotify({
+                title: "Quota for creating desktops full.",
+                    text: "Can't create another desktop, user quota full.",
+                    hide: true,
+                    delay: 3000,
+                    icon: 'fa fa-alert-sign',
+                    opacity: 1,
+                    type: 'error'
+                });
+        }else */
```
Comment (sim6) [inline webapp/webapp/webapp/static/admin/js/domains.js:54]: Why this commented block?
```suggestion
```

File: webapp/webapp/webapp/static/js/snippets/desktop_add.js [hunk 8]
```diff
@@ -0,0 +1,209 @@
...
+                }else{
+                    $('#modal_add_desktops').closest('.x_panel').addClass('datatables-error');
+                    $('#modalAddDesktop #datatables-error-status').html('No template selected').addClass('my-error');
+                }
+            }
+        });
+        
+        //~ $("#modalAddDesktop #btn-hardware").on('click', function(e){
+                //~ $('#modalAddDesktop #hardware-block').show();
+        //~ });
```
Comment (sim6) [inline webapp/webapp/webapp/static/js/snippets/desktop_add.js:169]: What about this comments?
```suggestion
```

File: webapp/webapp/webapp/static/js/snippets/desktop_add.js [hunk 9]
```diff
@@ -0,0 +1,209 @@
...
+     }       
+}
+
+function renderIcon1x(data){
+    return '<span class="xe-icon" data-pk="'+data.id+'">'+icon1x(data.icon)+'</span>'
+}
+
+function renderTemplateKind(data){
+    //~ if(data.kind=="public_template"){return "public";}
+    //~ if(data.kind=="user_template"){return "private";}
```
Comment (sim6) [inline webapp/webapp/webapp/static/js/snippets/desktop_add.js:206]: What about this comments?
```suggestion
```

**Review by sim6 — COMMENTED** [review]
Geat! It works for me, but please, could you remove the commented block?

**Review by sim6 — APPROVED** [review]
Thanks!

**Pull request merged**